azurerm/internal/services/containers/kubernetes_cluster_node_pool_data_source.go

```diff
@@ -276,6 +278,8 @@ func dataSourceKubernetesClusterNodePoolRead(d *schema.ResourceData, meta interf
 		}
 		d.Set("spot_max_price", spotMaxPrice)
 
+		d.Set("upgrade_settings", flattenUpgradeSettings(props.UpgradeSettings))
```

Comment:
since this is a complex object this wants to be:

```suggestion
		if err := d.Set("upgrade_settings", flattenUpgradeSettings(props.UpgradeSettings)); err != nil {
			return fmt.Errorf("setting `upgrade_settings`: %+v", err)
		}
```